### ISSUE TITLE
Unable to run libgdx on JDK 14

### ISSUE THREAD

**Issue by zskamljic** [libgdx/libgdx#5996]
#### Issue details
Libgdx crashes when trying to run or develop on JDK 14 with the stacktrace attached below. I assume lwjgl needs to be updated.

#### Reproduction steps/code
Simply create a new project with latest gradle wrapper (6.3). Try running the application, observe the crash.

#### Version of LibGDX and/or relevant dependencies
1.9.10

#### Stacktrace
```java
[LwjglApplication] Couldn't initialize audio, disabling audio
java.lang.UnsatisfiedLinkError: /tmp/libgdxtest/dd5c1a65/liblwjgl64.so: /usr/lib/jvm/java-14-openjdk/lib/libjawt.so: version `SUNWprivate_1.1' not found (required by /tmp/libgdxtest/dd5c1a65/liblwjgl64.so)
	at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
	at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2452)
	at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2508)
	at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2704)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2637)
	at java.base/java.lang.Runtime.load0(Runtime.java:745)
	at java.base/java.lang.System.load(System.java:1871)
	at org.lwjgl.Sys$1.run(Sys.java:70)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
	at org.lwjgl.Sys.doLoadLibrary(Sys.java:66)
	at org.lwjgl.Sys.loadLibrary(Sys.java:87)
	at org.lwjgl.Sys.<clinit>(Sys.java:117)
	at org.lwjgl.openal.AL.<clinit>(AL.java:59)
	at com.badlogic.gdx.backends.lwjgl.audio.OpenALAudio.<init>(OpenALAudio.java:72)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication.<init>(LwjglApplication.java:90)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication.<init>(LwjglApplication.java:71)
	at com.test.DesktopLauncher.main(DesktopLauncher.java:9)
```

#### Please select the affected platforms
- [ ] Android
- [ ] iOS (robovm)
- [ ] iOS (MOE)
- [ ] HTML/GWT
- [ ] Windows
- [x] Linux
- [ ] MacOS

**Comment by tommyettinger** [comment]
Uh... It's harder to evaluate this because you didn't supply any info about your OS, and you checked all the affected platforms boxes even though Android and GWT, at the very least, don't have Java 14 as an option for them. I'm guessing you're on some distro of Linux or on a Mac, based on the file path given in one place. I just ran an in-progress game of mine on AdoptOpenJDK 14, with LWJGL2 as you have it, on Windows 7 and on Manjaro Linux, and both ran it fine. https://imgur.com/DzMLTCG.png <-- This shows I'm using Gradle 6.3 and Java 14; I have modules set up for LWJGL2 and LWJGL3 and both run. It's possible your JDK has some bug present in it, or for some other reason doesn't provide the normal API. This has affected LWJGL2 in the past, but only on some (almost always Linux or Amazon Corretto) JDKs. AdoptOpenJDK builds of the JDK have always worked for me: https://adoptopenjdk.net/

**Comment by zskamljic** [comment]
You're correct, I'm on Linux, using openjdk: 
```openjdk 14 2020-03-17
OpenJDK Runtime Environment (build 14+36-1461)
OpenJDK 64-Bit Server VM (build 14+36-1461, mixed mode, sharing)```

I've also updated the targets accordingly. I have used LWJGL3 backend.

**Comment by tommyettinger** [comment]
There's absolutely a bug that arises somewhere from the combination of LWJGL2 and specific OpenJDK implementations, and I think you made the right move by switching to LWJGL3. You may encounter weirdness on some hardware with Linux and LWJGL3; there's an FPS regulation difference between 2 and 3, and some LWJGL3 games run at unrestricted FPS on some Linux machines. There are solutions (see https://github.com/libgdx/libgdx/issues/5826#issuecomment-578456720 ) if it's a problem for your case. I'm still pushing for more people to use AdoptOpenJDK while this kind of incompatibility with LWJGL2 persists on some JDKs.

**Comment by Torbuntu** [comment]
I've noticed issues with the packaged open jdk on fedora (idk 12 and up) so I moved to AdoptOpenjdk and haven't had any issues.

**Comment by xGnoSiSx** [comment]
@tommyettinger I was reading your comment and was alarmed for a moment. I know that libgdx uses lwjgl that has bindings to the native opengl implementation, so the JDK does not get involved at all with it!

Any issues launching, should not be attributed to libgdx, but on the JDK trying to start without proper java opengl support (*not* jwjgl bindings)

Newest JDKs have this issue, especially on OSX, where the JDK does not even start due to the opengl situation with apple, and there's project Lanai (https://wiki.openjdk.java.net/pages/viewpage.action?pageId=49250445), because the 2d stuff in Java must somehow work.

The question on Linux depends on what distro was involved and what opengl binaries were present in that installation.

**Comment by tommyettinger** [comment]
I've seen roughly the same error posted with an Ubuntu default JDK11 JVM, a more recent Amazon Corretto JVM, and if I remember it right there was a Fedora case as well (oh hi Torbuntu!). All mention `ld.so`, so somewhere in native code, and a linker error. I'm inclined to believe this has to do with some common but incorrect build setting when assembling OpenJDK; the JDKs can run at least some other applications. It's absolutely possible that OpenGL is related to the error, or that some JDKs are already removing support for some features in `sun.misc.Unsafe` or other internal APIs marked for eventual removal. There does appear to be a difference between LWJGL2 and LWJGL3 here, and version 3 seems to include a fix for whatever it is. The switch to GLFW is my guess why that fix works.

Personally, I'd push for halting support for MacOS entirely if or when OpenGL becomes unusable on that OS. Obviously iOS is a major target for mobile, via MobiVM/RoboVM, but for desktop OS targets... The game Halfway used libGDX amazingly well, and though their genre isn't the same as most other games you could compare with, 4% of sales being to MacOS is a pretty miserable return considering 50% of all support requests came from Macs. 
Here's a link to Robotality's [Halfway Post-Mortem](http://robotality.com/blog/wp-content/uploads/2015/07/HalfwayInPixelsFinal005.png).

**Comment by xGnoSiSx** [comment]
@tommyettinger Great info!

**Comment by NathanSweet** [comment]
Fixed by: 778c936c9cc35c20ded87c2ff95252c217831963
